Sources/BraintreeCore/BTHTTP.swift

```diff
@@ -137,63 +132,32 @@ class BTHTTP: NSObject, NSCopying, URLSessionDelegate {
 
     // MARK: - HTTP Method Helpers
 
-    func httpRequestWithCaching(
+    func httpRequest(
         method: String,
         path: String,
         parameters: [String: Any]? = [:],
+        shouldCache: Bool = false,
         completion: RequestCompletion?
     ) {
         createRequest(method: method, path: path, parameters: parameters) { request, error in
             guard let request = request else {
                 self.handleRequestCompletion(data: nil, request: nil, shouldCache: false, response: nil, error: error, completion: completion)
                 return
             }
-
-            var cachedResponse: CachedURLResponse? = URLCache.shared.cachedResponse(for: request) ?? nil
-
-            if self.cacheDateValidator.isCacheInvalid(cachedResponse ?? nil) {
-                URLCache.shared.removeAllCachedResponses()
-                cachedResponse = nil
-            }
-
-            // The increase in speed of API calls with cached configuration caused an increase in "network connection lost" errors.
-            // Adding this delay allows us to throttle the network requests slightly to reduce load on the servers and decrease connection lost errors.
-            DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                if let cachedResponse = cachedResponse {
```

Comment:
Is this the main difference in this PR? 
Are there still concerns about "network connection lost" errors?

Reply:
> TBH - I am still struggling to understand how this 100ms delay fixed anything. What it was doing was adding 100ms of delay to v1/configuration calls, as well as 100ms of delay if the config was cached. It was not adding 100ms delay to other API calls.

This is correct, the issue was not subsequent API calls, the issue was that when we started caching properly, we were returning from the venmo app without the loading screen (older version) _before_ we got the response back from the config API call. This resulted in the config network connection being lost during the switch/return.

Reply:
Hm, I'm still a little confused. So the config network call is what was seeing "network connection lost"? But it sounds like in this error case, the config was cached prior to launching Venmo, so why were we doing another config network call?

It sounds like a concurrency issue (Steven alluded to [here](https://github.com/braintree/braintree_ios/pull/807#pullrequestreview-942345074)), and the problem was that we weren't always properly waiting for a config value to be returned (whether from cache or network) before launching Venmo. My guess is multiple threads were updating the cache & removing items from it at the same time, and we "thought" it had a value in it, but by the time Venmo finished, another thread already wiped it.

Reply:
Curious - how did you run those 100 tests? I imagine you used some automated tool in Xcode to run them in parallel?

Reply:
Since the Venmo app can only be installed on physical devices all of our testing was manual. We basically both ran as many tests as quick as possible, so as soon as we returned and got a nonce we kicked off the flow again. At the time the Venmo team was not able to help with testing and we got very little information from the merchant on any device or version specifics.

Reply:
Resolving this comment in favor of discussion in PR description. Victoria, if you have more open questions on the original issue please let me know and re-open the thread!